src/v/model/model.cc

```diff
@@ -582,27 +584,74 @@ std::istream& operator>>(std::istream& is, recovery_validation_mode& vm) {
     return is;
 }
 
+iceberg_mode iceberg_mode::disabled
+  = iceberg_mode::make<iceberg_mode::variant::disabled>();
+iceberg_mode iceberg_mode::key_value
+  = iceberg_mode::make<iceberg_mode::variant::key_value>();
+iceberg_mode iceberg_mode::value_schema_id_prefix
+  = iceberg_mode::make<iceberg_mode::variant::value_schema_id_prefix>();
+
+void write(iobuf& out, const iceberg_mode& m) {
+    using serde::write;
+    write(out, m.kind());
+    if (m.kind() == iceberg_mode::variant::latest_protobuf_value) {
+        write(out, m.protobuf_full_name());
+    }
+}
+
+void read_nested(
+  iobuf_parser& in, iceberg_mode& m, const std::size_t bytes_left_limit) {
+    using serde::read_nested;
+    iceberg_mode::variant v = iceberg_mode::variant::disabled;
+    read_nested(in, v, bytes_left_limit);
+    switch (v) {
+    case iceberg_mode::variant::disabled:
+        m = iceberg_mode::disabled;
+        break;
+    case iceberg_mode::variant::key_value:
+        m = iceberg_mode::key_value;
+        break;
+    case iceberg_mode::variant::value_schema_id_prefix:
+        m = iceberg_mode::value_schema_id_prefix;
+        break;
+    case iceberg_mode::variant::latest_protobuf_value:
+        ss::sstring s;
+        read_nested(in, s, bytes_left_limit);
+        m = iceberg_mode::latest_protobuf_value(s);
+        break;
+    }
```

Comment:
wouldn't it be better to return inside the switch statement and if we get there to throw an exception? in the unlikely case we mess up somwhere/forget a feature gate we would end up with disabled iceberg.

this is backwards compatible with the old code I suppose?

Reply:
I think throwing is a good idea, will do.

Reply:
done